File: Easy/Pascal Triangle/pascal-triangle.java
//{ Driver Code Starts
// Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    // Driver code
    public static void main(String[] args) throws Exception {
        BufferedReader br =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().trim());
        while (t-- > 0) {
            int n = Integer.parseInt(br.readLine().trim());

            ArrayList<Long> ans = new Solution().nthRowOfPascalTriangle(n);
            printAns(ans);
        }
    }
    public static void printAns(ArrayList<Long> ans) {
        for (Long x : ans) {
            System.out.print(x + " ");
        }
        System.out.println();
    }
}

// } Driver Code Ends


// User function Template for Java

class Solution {
   ArrayList<Long> nthRowOfPascalTriangle(int n) {
         ArrayList<Long>  a = new ArrayList<>();
         long ans=1,mod=(long)1e9+7;
        for(int j=0;j<n;j++){
             a.add(ans);
             for(int i=j-1;i>=1;i--)
                 a.set(i,(a.get(i)+a.get(i-1))%mod);
         }return a;
}
}

File: Easy/Pascal Triangle/pascal-triangle.py
#User function Template for python3

#User function Template for python3
class Solution:
    def __init__(self):
        self.mod = 10 ** 9 + 7
    
    def modInverse(self, a):
        return pow(a, self.mod - 2, self.mod)
    
    def nthRowOfPascalTriangle(self, n):
        ans = [1] * n
        for i in range(1, n):
            ans[i] = (ans[i - 1] * (n - i) * self.modInverse(i)) % self.mod

        return ans

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__': 

	tc=int(input())
	while tc > 0:
	    n=int(input())
	    ob = Solution()
	    ans=ob.nthRowOfPascalTriangle(n)
	    for x in ans:
	        print(x, end=" ")
	    print()
	    tc=tc-1
# } Driver Code Ends